src/ucp/rndv/rndv_mtype.inl

```diff
@@ -22,7 +22,7 @@ ucp_proto_rndv_mtype_init(const ucp_proto_init_params_t *init_params,
     ucs_memory_type_t frag_mem_type = context->config.ext.rndv_frag_mem_type;
 
     if ((init_params->select_param->dt_class != UCP_DATATYPE_CONTIG) ||
-        (worker->mem_type_ep[mem_type] == NULL) ||
+        (mem_type == frag_mem_type) ||
```

Comment:
the first bullet is the scheme we want to support for MNNVL.
Regarding the check for memtype_ep, maybe we can add it as well but checking both `mem_type` and `frag_memtype`. Because we will also need to support the below scheme for pipelining legacy pinned memory thru fabric cuda memory, so this check will need to be modified anyway
`cuda->cuda(fabric)/cuda(fabric)->cuda`

Reply:
i think we need to make sure there is a memtype ep to comply between mem_type and frag_mem_type. it would probably share the logic with ucp_proto_rndv_mtype_ep function.